src/Mod/Part/Gui/ViewProviderBoolean.cpp

```diff
@@ -140,19 +201,32 @@ void ViewProviderBoolean::updateData(const App::Property* prop)
     }
 }
 
-bool ViewProviderBoolean::onDelete(const std::vector<std::string> &)
+bool ViewProviderBoolean::onDelete(const std::vector<std::string> &subNames)
 {
     // get the input shapes
     Part::Boolean* pBool = getObject<Part::Boolean>();
     App::DocumentObject *pBase = pBool->Base.getValue();
     App::DocumentObject *pTool = pBool->Tool.getValue();
 
-    if (pBase)
-        Gui::Application::Instance->showViewProvider(pBase);
-    if (pTool)
-        Gui::Application::Instance->showViewProvider(pTool);
-
-    return true;
+    // Prepare input objects list and description
+    std::vector<App::DocumentObject*> inputObjects;
+    if (pBase) inputObjects.push_back(pBase);
+    if (pTool) inputObjects.push_back(pTool);

```

Comment:
Since you're refactoring it, how about adding braces?
```suggestion
    if (pBase) {
        inputObjects.push_back(pBase);
    }
    if (pTool) {
        inputObjects.push_back(pTool);
    }
```